pkg/apis/core/validation/validation.go

```diff
@@ -6921,38 +6939,43 @@ func validateScopeSelector(resourceQuotaSpec *core.ResourceQuotaSpec, fld *field
 func ValidateResourceQuota(resourceQuota *core.ResourceQuota) field.ErrorList {
 	allErrs := ValidateObjectMeta(&resourceQuota.ObjectMeta, true, ValidateResourceQuotaName, field.NewPath("metadata"))
 
-	allErrs = append(allErrs, ValidateResourceQuotaSpec(&resourceQuota.Spec, field.NewPath("spec"))...)
-	allErrs = append(allErrs, ValidateResourceQuotaStatus(&resourceQuota.Status, field.NewPath("status"))...)
+	allErrs = append(allErrs, ValidateResourceQuotaSpec(&resourceQuota.Spec, nil, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidateResourceQuotaStatus(&resourceQuota.Status, nil, field.NewPath("status"))...)
```

Comment:
So you are saying that `ValidateResourceQuotaSpec` should not get the additional `oldResourceQuotaSpec *core.ResourceQuotaSpec` argument? It then cannot be called as part of an update.

Instead, we need the same code again in a `ValidateResourceQuotaSpecUpdate` where the additional special case of "name was already before update" is considered.

Isn't it better to avoid that duplication of code and add the `oldResourceQuotaSpec *core.ResourceQuotaSpec` parameter, as done in this PR at the moment?

Reply:
BTW, I think `ValidateResourceQuotaStatus` is useless here. The status is always going to be empty when `ValidateResourceQuota` is being called for a new resource, right?

Reply:
> Isn't it better to avoid that duplication of code and add the oldResourceQuotaSpec *core.ResourceQuotaSpec parameter, as done in this PR at the moment?

Clarified on Slack: some code duplication is worth it because of the extra clarity.